_inc/lib/class.core-rest-api-endpoints.php

```diff
@@ -619,6 +619,66 @@ public static function register_endpoints() {
 			)
 		);
 
+		register_rest_route(
+			'jetpack/v4',
+			'/recommendations/data',
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => __CLASS__ . '::get_recommendations_data',
+					'permission_callback' => __CLASS__ . '::update_settings_permission_check',
+				),
+				array(
+					'methods'             => WP_REST_Server::EDITABLE,
+					'callback'            => __CLASS__ . '::update_recommendations_data',
+					'permission_callback' => __CLASS__ . '::update_settings_permission_check',
+					'args'                => array(
+						'data' => array(
+							'required'          => true,
+							'type'              => 'object',
+							'validate_callback' => __CLASS__ . '::validate_recommendations_data',
+						),
+					),
+				),
+			)
+		);
+
+		register_rest_route(
+			'jetpack/v4',
+			'/recommendations/step',
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => __CLASS__ . '::get_recommendations_step',
+					'permission_callback' => __CLASS__ . '::update_settings_permission_check',
+				),
+				array(
+					'methods'             => WP_REST_Server::EDITABLE,
+					'callback'            => __CLASS__ . '::update_recommendations_step',
+					'permission_callback' => __CLASS__ . '::update_settings_permission_check',
+					'args'                => array(
+						'step' => array(
+							'required'          => true,
+							'type'              => 'string',
+							'validate_callback' => __CLASS__ . '::validate_string',
```

Comment:
Should we check against an array of possible values here instead?

Reply:
I had considered that but decided not to. The addition of new steps is the most likely future change so I wanted to keep that as easy as possible and since the client needs to contain the logic of the order of the different steps I wanted to keep the definition of the steps in the client only. I think validating the step as a string is enough for security purposes, but if there is another issue here that I haven't considered I would be happy to add validation against an array of possible values.